junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/EnabledCondition.java

```diff
@@ -0,0 +1,152 @@
+		String script = createScript(enabled);
+		logger.config(() -> "Script to evaluate: " + script);
+
+		// Find script engine and prepare bindings
+		ScriptEngine scriptEngine = findScriptEngine(enabled.engine());
+		logger.config(() -> "Found script engine: " + scriptEngine);
+
+		Bindings bindings = scriptEngine.getBindings(ScriptContext.ENGINE_SCOPE);
+		if (enabled.bindExtensionContext()) {
+			bindings.put("junit$context", context);
+		}
+		if (enabled.bindSystemProperties()) {
+			for (Map.Entry<Object, Object> it : System.getProperties().entrySet()) {
+				String key = "system$" + it.getKey().toString().replace('.', '_');
+				bindings.put(key, it.getValue());
+			}
+		}
+		logger.config(() -> "Bindings: " + bindings);
+
+		try {
+			Object result = scriptEngine.eval(script);
+			boolean ok = Boolean.parseBoolean(String.valueOf(result));
+			String reason = enabled.reason();
+			if (reason.isEmpty()) {
+				reason = String.format("Script `%s` evaluated to: %s", script, result);
+			}
+			return ok ? enabled(reason) : disabled(reason);
+		}
+		catch (ScriptException e) {
+			logger.config(e, () -> "Evaluation of @Enabled script failed, disabling execution");
+			return disabled(e.getMessage());
+		}
+	}
+
+	ScriptEngine findScriptEngine(String string) {
+		ScriptEngineManager manager = new ScriptEngineManager();
+		ScriptEngine scriptEngine = manager.getEngineByName(string);
+		if (scriptEngine == null) {
+			scriptEngine = manager.getEngineByExtension(string);
+		}
+		if (scriptEngine == null) {
+			scriptEngine = manager.getEngineByMimeType(string);
+		}
+		Preconditions.notNull(scriptEngine, "Script engine not found: " + string);
+		return scriptEngine;
+	}
+
+	String createScript(Enabled enabled) {
+		// trivial case: no imports, single script line
+		if (enabled.imports().length == 0 && enabled.value().length == 1) {
+			return enabled.value()[0];
+		}
+
+		// special case: "javascript"
+		boolean isJavaScript = "javascript".equals(enabled.engine());
+		if (isJavaScript) {
+			return createJavaScript(enabled);
+		}
+
+		// just join the script lines
+		return joinLines(enabled.delimiter(), Arrays.asList(enabled.value()));
+	}
+
+	private String createJavaScript(Enabled enabled) {
+		boolean injectJavaImporterStatements = enabled.imports().length > 0;
+
+		List<String> lines = new ArrayList<>();
+		if (injectJavaImporterStatements) {
+			String imports = String.join(", ", enabled.imports());
+			lines.add("var javaImporter = new JavaImporter(" + imports + ")");
+			lines.add("with (javaImporter) {");
+		}
+		for (String line : enabled.value()) {
+			if (injectJavaImporterStatements) {
+				line = "  " + line;
+			}
+			lines.add(line);
+		}
+		if (injectJavaImporterStatements) {
+			lines.add("}");
+			lines.add("");
+		}
+
+		return joinLines(enabled.delimiter(), lines);
+	}
+
+	private String joinLines(String delimiter, Iterable<? extends CharSequence> elements) {
+		if (delimiter.isEmpty()) {
+			delimiter = System.lineSeparator();
```

Comment:
Not sure we should make this dependent on the current platform. That way, test execution might yield different results on Windows and on the CI server. How about just using `\n`?

Reply:
Hm, what-if a script parser is implemented in platform-specific manner? I think the default is okay here, the user may override with `\n` or ` ` or _what-ever_ within the `@Conditional` usage.